### ISSUE TITLE
x/net/html: Subsequent Token method calls in Tokenizer don't return same result

### ISSUE THREAD

**Issue by rpeshkov** [golang/go#22621]
### What version of Go are you using (`go version`)?

go 1.9.2

### Does this issue reproduce with the latest release?

Yes, including the up-to-date x/net/html dependency

### What operating system and processor architecture are you using (`go env`)?

GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/rpeshkov"
GORACE=""
GOROOT="/usr/local/Cellar/go/1.9.2/libexec"
GOTOOLDIR="/usr/local/Cellar/go/1.9.2/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/dl/d4m5fbhs6k1cx9hhz1zdk4v40000gn/T/go-build962492023=/tmp/go-build -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"

### What did you do?

Link for code: https://play.golang.org/p/7ZwQB98kHE

### What did you expect to see?

```
<div class="hello">
<div class="hello">
```

### What did you see instead?

```
<div class="hello">
<>
```

**Comment by agnivade** [comment]
I believe your understanding of the API is incorrect.

[Next](https://godoc.org/golang.org/x/net/html#Tokenizer.Next) scans the next token and returns its type.

[Token](https://godoc.org/golang.org/x/net/html#Tokenizer.Next) returns the next token. So the code is doing what is expected.

Here is something which is more on the lines of what you are expecting - 

```go
func main() {
  reader := strings.NewReader("<div class=\"hello\">SomeText</div>")
  tokenizer := html.NewTokenizer(reader)
  tokenizer.Next()
  fmt.Printf("%s\n", tokenizer.Raw())
  fmt.Printf("%s\n", tokenizer.Raw())
}
```

**Comment by rpeshkov** [comment]
But `Raw` return slice of bytes, while Token returns an instance of Token struct.

It's ok that Next changes the internal state of the object, but I don't feel that it's kind of logical that Token changes the state.

Also, I think that documentation is a little bit incorrect. 

> Next scans the next token and returns its type.

That's correct

> Token returns the next token.

However, Token doesn't return the next token. It wraps the token you're currently on into struct and returns it to you. It doesn't somehow advance the iterator.

Basically, why I think that current logic is incorrect? Because since Token just wraps current 'state' into struct, it's weird for me that after wrap, internal state is messed.

**Comment by agnivade** [comment]
@namusyaka

**Comment by namusyaka** [comment]
@agnivade Thanks for pinging. I would like to work on this issue after resolving some crashing issues.
